src/main/rx/ghst.c

```diff
@@ -264,22 +280,55 @@ static bool ghstProcessFrame(const rxRuntimeState_t *rxRuntimeState)
                     break;
                 }
 
-                case GHST_UL_RC_CHANS_HS4_5TO8:     startIdx = 4;  break;
-                case GHST_UL_RC_CHANS_HS4_9TO12:    startIdx = 8;  break;
-                case GHST_UL_RC_CHANS_HS4_13TO16:   startIdx = 12; break;
+                case GHST_UL_RC_CHANS_HS4_5TO8:
+                case GHST_UL_RC_CHANS_HS4_12_5TO8:
+                    startIdx = 4;
+                    break;
+
+                case GHST_UL_RC_CHANS_HS4_9TO12:
+                case GHST_UL_RC_CHANS_HS4_12_9TO12:
+                    startIdx = 8;
+                    break;
+
+                case GHST_UL_RC_CHANS_HS4_13TO16:
+                case GHST_UL_RC_CHANS_HS4_12_13TO16:
+                    startIdx = 12;
+                    break;
+
                 default:
                     DEBUG_SET(DEBUG_GHST, DEBUG_GHST_UNKNOWN_FRAMES, ++unknownFrameCount);
                     break;
             }
 
-            if (startIdx > 0)
-            {
-                // remainder of uplink frame contains 4 more channels (8 bit), sent in a round-robin fashion
+            // We need to wait for the first RSSI frame to know ghstRfProtocol
+            if (ghstRfProtocol != GHST_RF_PROTOCOL_UNDEFINED) {
+                const ghstPayloadPulses_t* const rcChannels = (ghstPayloadPulses_t*)&ghstValidatedFrame->frame.payload;
+
+                // all uplink frames contain CH1..4 data (12 bit)
+                ghstChannelData[0] = rcChannels->ch1to4.ch1;
+                ghstChannelData[1] = rcChannels->ch1to4.ch2;
+                ghstChannelData[2] = rcChannels->ch1to4.ch3;
+                ghstChannelData[3] = rcChannels->ch1to4.ch4;
+
+                if (startIdx > 0) {
+                    // remainder of uplink frame contains 4 more channels (8 bit), sent in a round-robin fashion
+                    ghstChannelData[startIdx++] = rcChannels->cha << 2;
+                    ghstChannelData[startIdx++] = rcChannels->chb << 2;
+                    ghstChannelData[startIdx++] = rcChannels->chc << 2;
+                    ghstChannelData[startIdx++] = rcChannels->chd << 2;
+                }
+
+                // Recalculate old scaling to the new one
+                if(scalingLegacy) {
+                    for(int i = 0; i < 4; ++i) {
```

Comment:
```suggestion
                    for (int i = 0; i < 4; ++i) {
```

Reply:
Just a note: interesting why not initialize `i` with value 1 and use `i++`

Reply:
`++i` or `i++` does not change the logic. Initialization must be still the same.